library/src/main/java/io/realm/RealmBasedRecyclerViewAdapter.java

```diff
@@ -444,7 +444,7 @@ private void updateRowWrappers() {
     private RealmChangeListener getRealmChangeListener() {
         return new RealmChangeListener() {
             @Override
-            public void onChange() {
+            public void onChange(Object element) {
```

Comment:
Agree type should be changed but `Object` should be fine for now. Based on the [javadoc](https://realm.io/docs/java/latest/api/io/realm/RealmChangeListener.html) it may actually be the Realm obj that changed. 

Anyhow, I'll merge this for now and then fix this up later.